src/net/l0/net_tx.c

```diff
@@ -68,7 +68,12 @@ int net_tx(struct sk_buff *skb, struct net_header_info *hdr_info) {
 	dev = skb->dev;
 	assert(dev != NULL);
 
-	if (hdr_info->dst_p == NULL || (dev->flags & IFF_NOARP)) {
+	if (dev->flags & (IFF_LOOPBACK | IFF_NOARP)) {
+		eth_hdr(skb)->h_proto = htons(hdr_info->type);
+		goto send;
+	}
+
+	if (hdr_info->dst_p == NULL) {
 		/* it's local we don't need resolve hardware address */
```

Comment:
Is `hdr_info->dst_p == NULL` reachable now? You check if it is a loopback device above and go to send if so, it looks like this condition is unreachable anymore.

Reply:
It's  also for broadcast